seed/static/seed/js/controllers/inventory_list_controller.js

```diff
@@ -92,10 +94,73 @@ angular.module('BE.seed.controller.inventory_list', [])
       } else {
         // No profiles exist
         $scope.columns = _.reject(all_columns, 'is_extra_data');
-      }
+      };
 
       $scope.restoring = false;
 
+      // Find labels that should be displayed and organize by applied id
+      $scope.show_labels = [];
+      for (let n in labels) {
+        let label = labels[n];
+        if (label.show_in_list) {
+          for (let m in label.is_applied) {
+            let id = label.is_applied[m];
+            if (!$scope.show_labels[id]) {
+              $scope.show_labels[id] = [];
+            }
+            $scope.show_labels[id].push(label);
+          }
+        }
+      };
+
+      // Builds the html to display labels associated with this row entity
+      $scope.display_labels = function (entity) {
+        let id = $scope.inventory_type === 'properties' ? entity.property_view_id : entity.taxlot_view_id;
+        let labels = [];
+        let titles = [];
+        if ($scope.show_labels[id]) {
+          for (let i in $scope.show_labels[id]) {
+            let label = $scope.show_labels[id][i];
+            labels.push('<span class="', $scope.show_full_labels ? 'label' : 'label-bar', ' label-', label.label, '">', $scope.show_full_labels ? label.text : '', '</span>');
+            titles.push(label.text);
+          }
+        }
+        return ['<span title="', titles.join(', ') ,'" class="label-bars">', labels.join(''), '</span>'].join('');
+      };
+
+      $scope.show_full_labels = false;
+      $scope.toggle_labels = function () {
+        $scope.show_full_labels = !$scope.show_full_labels;
+        setTimeout(() => {
+          $scope.gridApi.grid.getColumn('labels').width = $scope.get_label_column_width();
+          let icon = document.getElementById('label-header-icon');
+          icon.classList.add($scope.show_full_labels ? 'fa-chevron-circle-left' : 'fa-chevron-circle-right');
+          icon.classList.remove($scope.show_full_labels ? 'fa-chevron-circle-right' : 'fa-chevron-circle-left');
+          $scope.gridApi.grid.refresh();
+        }, 0);
+      };
+
+      $scope.get_label_column_width = function () {
```

Comment:
Nice! I'm not sure if we want to extend the column as wide as the labels go though. I tested adding a lot of labels (which is a semi-reasonable thing, IDK what the max number of labels typically would be though), and it prevents you from viewing the data (cant scroll to the right either):
<img width="1440" alt="Screen Shot 2021-04-30 at 3 13 41 PM" src="https://user-images.githubusercontent.com/18518728/116755006-f42ada80-a9c6-11eb-9997-6c0880cfd5f2.png">

I reckon this could also happen with fewer labels with longer names.

I guess the alternative is to set a max-width and let the height of the row expand, or to set a max-width and make the td scrollable. I think either option is good

Reply:
I set a max width of 1000px (which should rarely if ever be hit) and set the container to allow scrolling in the x direction.    If this ends up being not enough for a user I would recommend turning some labels off.